cores/esp8266/IPAddress.h

```diff
@@ -24,21 +24,44 @@
 #include <WString.h>
 #include <Printable.h>
 
+#include <lwip/init.h>
+#include <lwip/ip_addr.h>
+
+#if LWIP_VERSION_MAJOR == 1
+// compatibility macros to make lwIP-v1 compiling lwIP-v2 API
+#define LWIP_IPV6_NUM_ADDRESSES 0
+#define ip_2_ip4(x) (x)
+#define ipv4_addr ip_addr
+#define IP_IS_V4_VAL(x) (1)
+#define IP_SET_TYPE_VAL(x,y) do { (void)0; } while (0)
```

Comment:
These macros are used where necessary, because of lwIP-v2 and IPv6 enabled. They are lwIP-v2's and do not exist in lwIP-v1.4. So I added them here to remain compatible with lwIP-v1.4.